<a href="https://colab.research.google.com/github/duellal/DS-Unit-1-Sprint-2-Statistics/blob/master/Duell_LS_DS_121_Statistics_Probability_Assignment_Dem_Rep.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

## Practice 1-Sample T-tests

One Sample t-tests determine whether or not a sample mean is statistically different from some known (or hypothesized) population mean. 

### 1) Load the Data
- Use the [automobile dataset](https://archive.ics.uci.edu/ml/datasets/Automobile)
- Fix the column headers
- Make sure NaNs are used to indicate missing values

Feel free to add code cells and text cells as needed throughout the assignment.

In [ ]:
import pandas as pd
from scipy import stats
import numpy as np

In [ ]:
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data

--2020-12-20 17:50:10--  https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data
Resolving archive.ics.uci.edu (archive.ics.uci.edu)... 128.195.10.252
Connecting to archive.ics.uci.edu (archive.ics.uci.edu)|128.195.10.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 25936 (25K) [application/x-httpd-php]
Saving to: ‘imports-85.data’

imports-85.data     100%[===================>]  25.33K   168KB/s    in 0.2s    

2020-12-20 17:50:11 (168 KB/s) - ‘imports-85.data’ saved [25936/25936]



In [ ]:
col_head = ['Symboling', 'Normalized-Losses', 'Make', 'Fuel-Type', 'Aspiration', 'Num-of-Doors', 
            'Body-Style', 'Drive-Wheels', 'Engine-Location', 'Wheel-base', 'Length', 'Width', 
            'Height', 'Curb-Weight', 'Engine-Type', 'Num-of-Cylinders', 'Engine-Size', 'Fuel-System', 
            'Bore', 'Stroke', 'Compression-Ratio', 'Horsepower', 'Peak-RPM', 'City-MPG', 'Highway-MPG', 'Price']

df = pd.read_csv('imports-85.data', names=col_head)
print(df.shape)
df.head()

(205, 26)


Symboling Normalized-Losses         Make  ... City-MPG Highway-MPG  Price
0          3                 ?  alfa-romero  ...       21          27  13495
1          3                 ?  alfa-romero  ...       21          27  16500
2          1                 ?  alfa-romero  ...       19          26  16500
3          2               164         audi  ...       24          30  13950
4          2               164         audi  ...       18          22  17450

[5 rows x 26 columns]

In [ ]:
df = df.replace({'?':np.NaN})

In [ ]:
df.isnull().sum()

Symboling             0
Normalized-Losses    41
Make                  0
Fuel-Type             0
Aspiration            0
Num-of-Doors          2
Body-Style            0
Drive-Wheels          0
Engine-Location       0
Wheel-base            0
Length                0
Width                 0
Height                0
Curb-Weight           0
Engine-Type           0
Num-of-Cylinders      0
Engine-Size           0
Fuel-System           0
Bore                  4
Stroke                4
Compression-Ratio     0
Horsepower            2
Peak-RPM              2
City-MPG              0
Highway-MPG           0
Price                 4
dtype: int64

In [ ]:
df.head()

Symboling Normalized-Losses         Make  ... City-MPG Highway-MPG  Price
0          3               NaN  alfa-romero  ...       21          27  13495
1          3               NaN  alfa-romero  ...       21          27  16500
2          1               NaN  alfa-romero  ...       19          26  16500
3          2               164         audi  ...       24          30  13950
4          2               164         audi  ...       18          22  17450

[5 rows x 26 columns]

### 2) Pretend that this dataset represents the cars at a used car lot in your local town. 

- Use df.sample() to pick a random sample of 10 cars. Note that because this sample is random we are going to set the `random_state` so that all of us in the class get the same random sample. Please set your random state to `30` when using `df.sample()`

Is your sample reflective of the population value in regards to highway-mpg? Find the mean for `highway-mpg` for the entire dataset and use a 1-sample t-test to compare your estaimated sample mean to the population mean. Can you say that they are different? 

In [ ]:
samp_df = df.sample(10, random_state=30)

In [ ]:
samp_df['Highway-MPG'].mean()

32.8

Null Hypothesis: The mean highway MPG is 32.8.

Alternative Hypothesis: The mean highway MPG is not 32.8.

Confidence Level: 95%

In [ ]:
stats.ttest_1samp(df['Highway-MPG'], 32.8)

Ttest_1sampResult(statistic=-4.259684116248613, pvalue=3.124442782308114e-05)

Based on the t-statistic of -4.26 and a P-value of 0.00000312, there is a 0.000312% chance that the null hypothesis is true, therefore since it is below 5%, I will reject my null hypothesis and assume that the mean highway MPG is not 32.8.

### The salesman says the cars he sells typically have a fuel efficiency of about 35 miles per gallon on the highway. You want to verify his claim but can only test 10 cars. Using your sample of 10, test his claim and report your results.

Null Hypothesis: Cars sold by the salesman will have an average fuel efficency of 35 MPG on the highway.

Alternative Hypothesis: Cars sold by the salesman will not have an average fuel efficiency of 35 MPG on the highway.

Condifence Level: 95%

In [ ]:
stats.ttest_1samp(samp_df['Highway-MPG'], 35)

Ttest_1sampResult(statistic=-0.9570244044334747, pvalue=0.36355472283248624)

The t-statistic is -0.957 and the p-value is 0.363. There is a 36.3% chance that the null hypothesis is true. Therefore, I cannot reject the null hypothesis and assume that it is true. 

###Imagine that you now have the capacity to test 100 cars on the lot. Using the same random state of `30`, take a sample of 100 cars. Run a t-test to verify the salesman's claim with your new larger sample. Do you reach the same conclusion as you did with the sample size of 10? 

In [ ]:
samp_df2 = df.sample(100, random_state=30)

Null Hypothesis: Cars sold by the salesman will have an average fuel efficency of 35 MPG on the highway.

Alternative Hypothesis: Cars sold by the salesman will not have an average fuel efficiency of 35 MPG on the highway.

Condifence Level: 95%

In [ ]:
stats.ttest_1samp(samp_df2['Highway-MPG'], 35)

Ttest_1sampResult(statistic=-6.5185242467032305, pvalue=3.003137417468754e-09)

The t-statistic is -6.518 and the p-value is 0.0000000003. The chance that the null hypothesis is true is 0.00000003%, therefore I can reject the null hypothesis and assume that the cars sold by the salesman will not have an average efficiency of 35 MPG on the highway.

Why might these two t-tests using the same dataset lead to different conclusions about the salesman's claim?

The two t-tests using the same dataset lead to different conclusions because the first sample was not a very big sample size, and the second sample was significantly bigger than the first. So, due to the law of large numbers in statistics, the second sample for the t-test lead to a more accurate conclusion.

## Practice 2-Sample T-tests

Two Sample t-tests determine whether or not two sample means are statistically different from each other. 

This portion of your assignment requires you to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). 


### 1) Load the data

The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

- Read the dataset in from UCI, you'll need to provide a list of column headers
- Encode "yes" votes as 1 and "no" votes as 0. (You can use `df.replace()` to swap out these values)
- Use dataframe filtering to split the dataframe into two new dataframes based on party. Hold all republicans in one dataframe and all democrats in the other. These will be our two different "samples."



In [ ]:
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data

--2020-12-20 17:50:11--  https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data
Resolving archive.ics.uci.edu (archive.ics.uci.edu)... 128.195.10.252
Connecting to archive.ics.uci.edu (archive.ics.uci.edu)|128.195.10.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 18171 (18K) [application/x-httpd-php]
Saving to: ‘house-votes-84.data’

house-votes-84.data 100%[===================>]  17.75K   118KB/s    in 0.2s    

2020-12-20 17:50:12 (118 KB/s) - ‘house-votes-84.data’ saved [18171/18171]



In [ ]:
column_headers = ['party', 'handicapped-infants', 'water-project', 'budget', 'physician-fee-freeze', 'el-salvador-aid',
                  'religious-groups', 'anti-satellite-ban', 'aid-to-contras', 'mx-missile', 'immigration',
                  'synfuels', 'education', 'right-to-sue', 'crime', 'duty-free', 'south-africa']


df = pd.read_csv('house-votes-84.data', names=column_headers)
print(df.shape)
df.head()

(435, 17)


party handicapped-infants water-project  ... crime duty-free south-africa
0  republican                   n             y  ...     y         n            y
1  republican                   n             y  ...     y         n            ?
2    democrat                   ?             y  ...     y         n            n
3    democrat                   n             y  ...     n         n            y
4    democrat                   y             y  ...     y         y            y

[5 rows x 17 columns]

In [ ]:
df = df.replace({'y': 1, 'n': 0, '?':np.NaN})

In [ ]:
dem = df[df['party']=='democrat']
rep = df[df['party']=='republican']

dem.head()

party  handicapped-infants  water-project  ...  crime  duty-free  south-africa
2  democrat                  NaN            1.0  ...    1.0        0.0           0.0
3  democrat                  0.0            1.0  ...    0.0        0.0           1.0
4  democrat                  1.0            1.0  ...    1.0        1.0           1.0
5  democrat                  0.0            1.0  ...    1.0        1.0           1.0
6  democrat                  0.0            1.0  ...    1.0        1.0           1.0

[5 rows x 17 columns]

### 2) Perform two sample T-tests on different issues and report the results.

- Find an issue that democrats support more than republicans with p < 0.01 (significant at the 99% level).
- Find an issue that republicans support more than democrats with p < 0.01 (significant at the 99% level).
- Find an issue where the difference between republicans and democrats has p > 0.1 (Not significant at the 90% level - i.e. there may not be much of a difference the two sample means)

Please for each test that you run state your null and alternative hypothesis and then write a conclusion reflecting on the null and alternative hypothesis.

Remember, that two-sample t-tests will only tell us if the two groups are *different* from one another. We'll have to look at their sample means directly or use the sign on the t-statistic to know which of the two sample means is larger. 

#Performing T-test on all 16 issues:

##Handicapped Infants:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['handicapped-infants'], rep['handicapped-infants'], nan_policy='omit')

Ttest_indResult(statistic=9.205264294809222, pvalue=1.613440327937243e-18)

The t-test is 9.205 and the p-value is 0.0000000000000000016. The chance that the null hypothesis is true is 0.00000000000000016%, therefore since it does not match my confidence level, I will reject my null hypothesis and assume that the alternative hypothesis is true: the sample means are different.

In [ ]:
print(dem['handicapped-infants'].mean())
print(rep['handicapped-infants'].mean())

0.6046511627906976
0.18787878787878787


However, the t-statistic is positive (leaning more toward democrats) and the mean of democrats supporting this bill is higher, we can conclude that democrats do support the handicapped infants bill more than republicans.

###Water Project:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['water-project'], rep['water-project'], nan_policy='omit')

Ttest_indResult(statistic=-0.08896538137868286, pvalue=0.9291556823993485)

In [ ]:
print(dem['water-project'].mean())
print(rep['water-project'].mean())

0.502092050209205
0.5067567567567568


The t-test is -0.0889 and the p-value is 0.929. The chance that the null hypothesis is true is 92.9%, therefore since it is not 99% or higher, I will reject my null hypothesis and assume that the alternative hypothesis is true.

However, the t-statistic is in the negative, meaning the votes are more republicans. However, it is a small value leaning toward republicans, and if we look at the means for democrats and republicans, we can see that they support the water project about the same.

###Budget: 

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['south-africa'], rep['south-africa'], nan_policy='omit')

Ttest_indResult(statistic=6.849454815841208, pvalue=3.652674361672226e-11)

In [ ]:
print(dem['budget'].mean())
print(rep['budget'].mean())

0.8884615384615384
0.13414634146341464


The t-statistic is 6.849 and the p-value is 0.0000000000355. Therefore the chance that the null hypothesis is true is 0.00000000355%, I will reject my null hypothesis and assume that the alternative hypothesis is true: the sample means are different.

However, since the t-statistic is postive and when looking at the means for republicans and democrats who support this bill, we can conclude that democrats support budget more than republicans. 

##Physician Fee Freeze:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['physician-fee-freeze'], rep['physician-fee-freeze'], nan_policy='omit')

Ttest_indResult(statistic=-49.36708157301406, pvalue=1.994262314074344e-177)

In [ ]:
print(dem['physician-fee-freeze'].mean())
print(rep['physician-fee-freeze'].mean())

0.05405405405405406
0.9878787878787879


The t-statistic is -49.367 and the p-value is 0. Therefore the chance that the null hypothesis is true is 0%, I will reject my null hypothesis and assume that the alternative hypothesis is true: the sample means are different. 

Looking at the t-statistic and the sample means for republicans and democrats, we can conclude that the republicans support the physician fee freeze more than republicans.

##El Salvador Aid:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['el-salvador-aid'], rep['el-salvador-aid'], nan_policy='omit')

Ttest_indResult(statistic=-21.13669261173219, pvalue=5.600520111729011e-68)

In [ ]:
print(dem['el-salvador-aid'].mean())
print(rep['el-salvador-aid'].mean())

0.21568627450980393
0.9515151515151515


The t-statistic is -21.137 and the p-value is 0. Therefore the chance that the null hypothesis is true is 0%, I will reject my null hypothesis and assume that the alternative hypothesis is true: the sample means are different.

Looking at the t-statistic and the sample means for republicans and democrats, we can conclude that republicans do support El Salvador aid more than democrats.

###Religious Groups:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['religious-groups'], rep['religious-groups'], nan_policy='omit')

Ttest_indResult(statistic=-9.737575825219457, pvalue=2.3936722520597287e-20)

In [ ]:
print(dem['religious-groups'].mean())
print(rep['religious-groups'].mean())

0.47674418604651164
0.8975903614457831


The t-statistic is -9.737 and the p-value is 0.0000000000000000000239. The chance that the null hypothesis is true is 0.00000000000000000239%. Therefore I will reject the null hypothesis and assume that the sample means are different.

Looking at the t-statistic and the sample means, we can conclude that republicans do support religious groups more than democrats.

##Anti Satellite Ban:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['anti-satellite-ban'], rep['anti-satellite-ban'], nan_policy='omit')

Ttest_indResult(statistic=12.526187929077842, pvalue=8.521033017443867e-31)

In [ ]:
print(dem['anti-satellite-ban'].mean())
print(rep['anti-satellite-ban'].mean())

0.7722007722007722
0.24074074074074073


The t-statistic is 12.526 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means are different.

However, looking at the t-statistic and the sample means, we can conclude that democrats do support the anti-satellite ban more than republicans.

##Aid to Contras:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['aid-to-contras'], rep['aid-to-contras'], nan_policy='omit')

Ttest_indResult(statistic=18.052093200819733, pvalue=2.82471841372357e-54)

In [ ]:
print(dem['aid-to-contras'].mean())
print(rep['aid-to-contras'].mean())

0.8288973384030418
0.15286624203821655


The t-statistic is 18.052 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means are different.

However, looking at the t-statistic and the sample means, we can conclude that democrats do support aid to contras more than republicans.

##Mx Missile:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['mx-missile'], rep['mx-missile'], nan_policy='omit')

Ttest_indResult(statistic=16.437503268542994, pvalue=5.03079265310811e-47)

In [ ]:
print(dem['mx-missile'].mean())
print(rep['mx-missile'].mean())

0.7580645161290323
0.11515151515151516


The t-statistic is 16.438 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means are different. 

Looking at the t-statistic and the sample means, we can conclude that the democrats support mx missile more than republicans.

##Immigration:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['immigration'], rep['immigration'], nan_policy='omit')

Ttest_indResult(statistic=-1.7359117329695164, pvalue=0.08330248490425066)

In [ ]:
print(dem['immigration'].mean())
print(rep['immigration'].mean())

0.4714828897338403
0.5575757575757576


The t-statistic is -1.736 and the p-value is 0.083. The chance that the null hypothesis is true is 8.3%. Therefore I will reject the null hypothesis and assume that the sample means are different.

Looking at the t-statistic and the sample means, we can conclude that the republicans support immigration a bit more than democrats.

##Synfuels:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['synfuels'], rep['synfuels'], nan_policy='omit')

Ttest_indResult(statistic=8.293603989407588, pvalue=1.5759322301054064e-15)

In [ ]:
print(dem['synfuels'].mean())
print(rep['synfuels'].mean())

0.5058823529411764
0.1320754716981132


The t-statistic is 8.294 and the p-value is 0.00000000000000158. The chance that the null hypothesis is true is 0.0000000000000158%. Therefore I will reject the null hypothesis and assume that the sample means are different.

Looking at the t-statistic and the sample means, we can conclude that democrats support synfuels more than republicans.

##Education:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['education'], rep['education'], nan_policy='omit')

Ttest_indResult(statistic=-20.500685724563073, pvalue=1.8834203990450192e-64)

In [ ]:
print(dem['education'].mean())
print(rep['education'].mean())

0.14457831325301204
0.8709677419354839


The t-statistic is -20.5 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means are different.

Looking at the t-statistic and the sample means, we can conlude that republicans support education more than democrats.

##Right to Sue:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['right-to-sue'], rep['right-to-sue'], nan_policy='omit')

Ttest_indResult(statistic=-13.51064251060933, pvalue=1.2278581709672758e-34)

In [ ]:
print(dem['right-to-sue'].mean())
print(rep['right-to-sue'].mean())

0.2896825396825397
0.8607594936708861


The t-statistic is -13.51 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means will be different.

Looking at the t-statistic and the sample means, we can conclude that republicans support right to sue more than democrats.

##Crime:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['crime'], rep['crime'], nan_policy='omit')

Ttest_indResult(statistic=-16.342085656197696, pvalue=9.952342705606092e-47)

In [ ]:
print(dem['crime'].mean())
print(rep['crime'].mean())

0.35019455252918286
0.9813664596273292


The t-statistic is -16.342 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means will be different.

Looking at the t-statistic and the sample means, we can conclude that republicans support crime more than democrats.

##Duty Free:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['duty-free'], rep['duty-free'], nan_policy='omit')

Ttest_indResult(statistic=12.853146132542978, pvalue=5.997697174347365e-32)

In [ ]:
print(dem['duty-free'].mean())
print(rep['duty-free'].mean())

0.6374501992031872
0.08974358974358974


The t-statistic is 12.853 and the p-value is 0. The chance that the null hypothesis is true is 0%. Therefore I will reject the null hypothesis and assume that the sample means will be different.

Looking at the t-statistic and the sample means, we can conlude that democrats support duty free more than republicans.

##South Africa:

Null Hypothesis: The sample means will be equal.

Alternative Hypothesis: The sample means will be different.

Confidence Level: 99%

In [ ]:
stats.ttest_ind(dem['south-africa'], rep['south-africa'], nan_policy='omit')

Ttest_indResult(statistic=6.849454815841208, pvalue=3.652674361672226e-11)

In [ ]:
print(dem['south-africa'].mean())
print(rep['south-africa'].mean())

0.9351351351351351
0.6575342465753424


The t-statistic is 6.849 and the p-value is 0.0000000000365. The chance that the null hypothesis is true is 0.00000000365%. Therefore I will reject the null hypothesis and assume that the sample means will be different.

Looking at the t-statistic and the sample means, we can conclude that democrats support south africa more than republicans.

#Conclusion:

Democrats support more than republicans with p < 0.01 (significant at the 99% level):
*  Handicapped Infants
*  Budget
*  Anti Satellite Ban
*  Aid to Contras
*  Mx Missile
*  Synfuels
*  Duty Free
*  South Africa

Republicans support more than democrats with p < 0.01 (significant at the 99% level):
*  Physician Fee Freeze
*  El Salvador Aid
*  Religious Groups
*  Immigration
*  Education
*  Right to Sue
*  Crime

Difference between republicans and democrats has p > 0.1 (Not significant at the 90% level - i.e. there may not be much of a difference the two sample means):
*  Water Project

# Stretch Goals:

### 1) Use functions and some iterator (for loop, .apply(), list comprehension, etc.) to perform two sample t-tests on every issue in the dataset in an automated fashion.



### 2) Work on Performing a T-test without using Scipy in order to get "under the hood" and learn more thoroughly about this topic.
### Start with a 1-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://blog.minitab.com/hs-fs/hubfs/Imported_Blog_Media/701f9c0efa98a38fb397f3c3ec459b66.png?width=247&height=172&name=701f9c0efa98a38fb397f3c3ec459b66.png) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

### Be sure to check your work using Scipy!





In [ ]:
### YOUR WORK HERE

### 3) Then try a 2-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://lh3.googleusercontent.com/proxy/rJJ5ZOL9ZDvKOOeBihXoZDgfk7uv1YsRzSQ1Tc10RX-r2HrRpRLVqlE9CWX23csYQXcTniFwlBg3H-qR8MKJPBGnjwndqlhDX3JxoDE5Yg) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Be sure to check your work using Scipy!

In [ ]:
### YOUR WORK HERE